aten/src/ATen/native/cuda/layer_norm_kernel.cu

```diff
@@ -504,7 +504,7 @@ __global__ void GammaBetaBackwardCUDAKernel_32x32(
 
   if (j < N) {
     constexpr int unroll_factor = 8;
-    int laneId = threadIdx.x & 0x1f;
+    int laneId = threadIdx.x % C10_WARP_SIZE;
```

Comment:
Do you know if nvcc guarantees equivalence between `(foo & 0x1f)` and `(foo % 32)`? I think behavior is slightly different if `foo` is negative. Please consider, for consistency sake to use `& (size-1)`
```suggestion
    int laneId = threadIdx.x & (C10_WARP_SIZE -1);
```